sup3r/bias/bias_calc.py

```diff
@@ -581,12 +596,30 @@ def fill_extend(self, out, smooth_extend):
         for key, arr in out.items():
             nan_mask = np.isnan(arr[..., 0])
             for idt in range(self.NT):
-                arr[..., idt] = nn_fill_array(arr[..., idt])
+
+                arr_smooth = arr[..., idt]
+
+                needs_fill = (fill_extend or smooth_extend > 0
+                              or smooth_interior > 0)
+
+                if needs_fill:
+                    arr_smooth = nn_fill_array(arr_smooth)
```

Comment:
Probably more accurate to do a `if np.isnan(arr_smooth).any() and fill_extend:` instead of the funky OR boolean